pkg/engine/jsonContext.go

```diff
@@ -195,3 +199,22 @@ func fetchConfigMap(entry kyverno.ContextEntry, lister dynamiclister.Lister) ([]
 
 	return data, nil
 }
+
+// mapRefactorizer recursively iterates through a map and updates its values in the following way
+// whenever it encounters a string value containing "\n",
+// it converts it into a []string by splitting it by "\n"
+func mapRefactorizer(m map[string]interface{}) map[string]interface{} {
+	for k, v := range m {
+		switch typedValue := v.(type) {
+		case string:
+			if strings.Contains(typedValue, "\n") {
+				m[k] = strings.Split(typedValue, "\n")
+			}
```

Comment:
How do we handle the previous case?
```
[\"cluster-admin\", \"cluster-operator\", \"tenant-admin\"]
```

Reply:
Here, basically, `[\"cluster-admin\", \"cluster-operator\", \"tenant-admin\"]` will be interpreted as it is because, it doesn't contain any "\n" to be split.

Reply:
It seems like it would be necessary to ensure users can still specify an array of strings like Kyverno presently requires. Multi-line strings in YAML should be an option but not requirement.

Reply:
Yeah true, the current implementation of this PR ensures that sort of backwards compatibility.